elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -122,15 +146,61 @@ public void commit() {
         criterion = CriterionFilterOperation.andWithNull(criterion,
                 criterionFilterOperation.applyAll(filteredPredicates));
 
-        // if no criterion then return all objects
-        if (criterion == null) {
-            return loadObjects(loadClass);
+
+        final Pagination pagination = filterScope.hasPagination() ? filterScope.getRequestScope().getPagination()
+                : null;
+
+        // if we have sorting and sorting isn't empty, then we should pull dictionary to validate the sorting rules
+        Set<Order> validatedSortingRules = null;
+        if (filterScope.hasSortingRules()) {
+            final Sorting sorting = filterScope.getRequestScope().getSorting();
+            final Map<String, Sorting.SortOrder> validSortingRules = sorting.getValidSortingRules(loadClass,
```

Comment:
This was moved directly into the Sorting.java. I am not sure if we want to support data store dependent code in Sorting since it is part of elide.core, so I moved this here. I made a best effort to keep everything in elide.core 100% generic.